packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -414,6 +420,14 @@ export async function runNpmInstall(
       destPath
     );
 
+    if (!packageJsonPath) {
+      debug(
+        `Skipping dependency installation because no package.json was found for ${destPath}`
+      );
+      runNpmInstallSema.release();
+      return false;
+    }
```

Comment:
Note: this change was required because yarn somehow allowed us to run `yarn install`, even if there was no `package.json` but `npm install` will throw.

This might end up being a tiny perf improvement for the case when no package.json is found since we then won't need to invoke a package manager at all.